hudi-common/pom.xml

```diff
@@ -147,6 +147,16 @@
       <scope>test</scope>
     </dependency>
 
+    <!-- Spark -->
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-core_${scala.binary.version}</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-sql_${scala.binary.version}</artifactId>
+    </dependency>
+
```

Comment:
`hudi-common` is a base module and feels a little weird that it relies on spark-core/spark-sql, would we remove them and move `getAllDataFilesForMarkers` method to `hudi-client` module, wdyt? cc @bvaradar @vinothchandar

Reply:
Okay so is the suggestion to move this method over to `HoodieTable` in `hudi-client` module, considering that is the only place this method is used ?

Reply:
yes.. that ll do..